Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean

```diff
@@ -249,6 +250,92 @@ theorem abs_log_sub_add_sum_range_le {x : ℝ} (h : |x| < 1) (n : ℕ) :
   -- fourth step: conclude by massaging the inequality of the third step
   simpa [F, div_mul_eq_mul_div, pow_succ] using C
 
+-- see https://github.com/leanprover-community/mathlib4/issues/29041
+set_option linter.unusedSimpArgs false in
+/--
+Compute the derivative of the difference between `1/2 * log ((1+x)/(1-x))` and its
+Taylor series at `0` up to order `n`. This is an auxiliary lemma for
+`sum_range_sub_log_div_le` and `sum_range_le_log_div`.
+Note that thanks to the geometric series, the derivative has a particularly simple form, and means
+that it is more convenient to avoid Taylor's theorem.
+-/
+private lemma sum_range_sub_log_div_le_aux {y : ℝ} (n : ℕ) (hy₁ : -1 < y) (hy₂ : y < 1) :
+    HasDerivAt
+      (fun x ↦ 1 / 2 * log ((1 + x) / (1 - x)) - (∑ i ∈ range n, x ^ (2 * i + 1) / (2 * i + 1)))
+      ((y ^ 2) ^ n / (1 - y ^ 2)) y := by
+  refine ((((((hasDerivAt_id _).const_add _).div ((hasDerivAt_id _).const_sub _) (by grind)).log
+          ?_).const_mul _).sub (HasDerivAt.fun_sum fun i hi ↦ (hasDerivAt_pow _ _).div_const _))
+        |>.congr_deriv ?_
```

Comment:
Out of curiosity, can the `data_synth` tactic of Tomas Skrivan do this? This would need to be tested on a different branch because I think that hasn't been merged yet.